File: Flask/controller/GENERIC.py
from flask import *

class ControllerGeneric(object):

	@staticmethod
	def viewTest():
		# render a view
		return render_template('GENERIC/CRUD.html');

	@staticmethod
	def viewCreate():
		# render a view
		return render_template('GENERIC/create.html');


	@staticmethod
	def viewRead():
		# render a view
		return render_template('GENERIC/read.html');


	@staticmethod
	def viewReadOne():
		# render a view
		return render_template('GENERIC/readOne.html');


	@staticmethod
	def viewUpdate():
		# render a view
		return render_template('GENERIC/update.html');


	@staticmethod
	def viewDelete():
		# render a view
		return render_template('GENERIC/delete.html');


	# api routes

	@staticmethod
	def apiTest():
		# perform some API action
		data = {'key':'value'};
		return jsonify(data);

	@staticmethod
	def apiCreate():
		# perform some API action
		data = {'key':'value'};
		return jsonify(data);

	@staticmethod
	def apiRead():
		# perform some API action
		data = {'key':'value'};
		return jsonify(data);

	@staticmethod
	def apiReadOne():
		# perform some API action
		data = {'key':'value'};
		return jsonify(data);

	@staticmethod
	def apiUpdate():
		# perform some API action
		data = {'key':'value'};
		return jsonify(data);

	@staticmethod
	def apiDelete():
		# perform some API action
		data = {'key':'value'};
		return jsonify(data);

File: Sinatra/controller/GENERIC.rb
class ControllerGeneric

	def self.viewTest():
		# render a view
		erb :'GENERIC/CRUD.html'
	end

	def self.viewCreate():
		# render a view
		erb :'GENERIC/create.html'
	end

	def self.viewRead():
		# render a view
		erb :'GENERIC/read.html'
	end

	def self.viewReadOne():
		# render a view
		erb :'GENERIC/readOne.html'
	end

	def self.viewUpdate():
		# render a view
		erb :'GENERIC/update.html'
	end

	def self.viewDelete():
		# render a view
		erb :'GENERIC/delete.html'
	end

	# api routes

	def self.apiTest():
		# perform some API action
	end

	def self.apiCreate():
		# perform some API action
	end

	def self.apiRead():
		# perform some API action
	end

	def self.apiReadOne():
		# perform some API action
	end

	def self.apiUpdate():
		# perform some API action
	end

	def self.apiDelete():
		# perform some API action
	end

end